homeassistant/components/neato/__init__.py

```diff
@@ -1,197 +1,117 @@
-    "maint_brush_stuck": "Brush stuck",
-    "maint_brush_overload": "Brush overloaded",
-    "maint_bumper_stuck": "Bumper stuck",
-    "maint_customer_support_qa": "Contact customer support",
-    "maint_vacuum_stuck": "Vacuum is stuck",
-    "maint_vacuum_slip": "Vacuum is stuck",
-    "maint_left_drop_stuck": "Vacuum is stuck",
-    "maint_left_wheel_stuck": "Vacuum is stuck",
-    "maint_right_drop_stuck": "Vacuum is stuck",
-    "maint_right_wheel_stuck": "Vacuum is stuck",
-    "not_on_charge_base": "Not on the charge base",
-    "nav_robot_falling": "Clear my path",
-    "nav_no_path": "Clear my path",
-    "nav_path_problem": "Clear my path",
-    "nav_backdrop_frontbump": "Clear my path",
-    "nav_backdrop_leftbump": "Clear my path",
-    "nav_backdrop_wheelextended": "Clear my path",
-    "nav_mag_sensor": "Clear my path",
-    "nav_no_exit": "Clear my path",
-    "nav_no_movement": "Clear my path",
-    "nav_rightdrop_leftbump": "Clear my path",
-    "nav_undocking_failed": "Clear my path",
-}
-
-ALERTS = {
-    "ui_alert_dust_bin_full": "Please empty dust bin",
-    "ui_alert_recovering_location": "Returning to start",
-    "ui_alert_battery_chargebasecommerr": "Battery error",
-    "ui_alert_busy_charging": "Busy charging",
-    "ui_alert_charging_base": "Base charging",
-    "ui_alert_charging_power": "Charging power",
-    "ui_alert_connect_chrg_cable": "Connect charge cable",
-    "ui_alert_info_thank_you": "Thank you",
-    "ui_alert_invalid": "Invalid check app",
-    "ui_alert_old_error": "Old error",
-    "ui_alert_swupdate_fail": "Update failed",
-    "dustbin_full": "Please empty dust bin",
-    "maint_brush_change": "Change the brush",
-    "maint_filter_change": "Change the filter",
-    "clean_completed_to_start": "Cleaning completed",
-    "nav_floorplan_not_created": "No floorplan found",
-    "nav_floorplan_load_fail": "Failed to load floorplan",
-    "nav_floorplan_localization_fail": "Failed to load floorplan",
-    "clean_incomplete_to_start": "Cleaning incomplete",
-    "log_upload_failed": "Logs failed to upload",
-}
-
-
-def setup(hass, config):
+
+async def async_setup(hass, config):
     """Set up the Neato component."""
+
+    if NEATO_DOMAIN not in config:
+        # There is an entry and nothing in configuration.yaml
+        return True
+
+    entries = hass.config_entries.async_entries(NEATO_DOMAIN)
+    hass.data[NEATO_CONFIG] = config[NEATO_DOMAIN]
+
+    if entries:
+        # There is an entry and something in the configuration.yaml
+        entry = entries[0]
+        conf = config[NEATO_DOMAIN]
+        if (
+            entry.data[CONF_USERNAME] == conf[CONF_USERNAME]
+            and entry.data[CONF_PASSWORD] == conf[CONF_PASSWORD]
+            and entry.data[CONF_VENDOR] == conf[CONF_VENDOR]
+        ):
+            # The entry is not outdated
+            return True
+
+        # The entry is outdated
+        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
```

Comment:
This will remove all devices and entity registry entries, including any customizations that the user had. It's better to update the data in the config entry .

Reply:
done

Reply:
I have reverted this until we know how to validate the updated configuration. See below for details

Reply:
done